app/components/Welcome.js

```diff
@@ -19,25 +32,56 @@ export default class Welcome extends Component {
     setNotFirstTimeFlag()
   }
 
+  selectedLanguage (i) {
+    // set the language to the selected value
+    i18n.changeLanguage(this.state.options[i].code)
+    setDefaultLanguage(this.state.options[i].code)
+  }
+
+  onLanguageChange (event) {
+    // Gets the index of the selected option
+    this.selectedLanguage(event.target.selectedIndex)
+  }
+  renderOptions () {
+    let options = [...SUPPORTED_LANGS]
```

Comment:
Since you are not reasigning this options variable, consider using `const` instead